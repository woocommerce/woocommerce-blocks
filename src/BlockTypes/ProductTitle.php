<?php
namespace Automattic\WooCommerce\Blocks\BlockTypes;

/**
 * ProductTitle class.
 */
class ProductTitle extends AbstractBlock {

	/**
	 * Block name.
	 *
	 * @var string
	 */
	protected $block_name = 'product-title';

	/**
	 * API version name.
	 *
	 * @var string
	 */
	protected $api_version = '2';

	/**
	 * Get block supports. Shared with the frontend.
	 * IMPORTANT: If you change anything here, make sure to update the JS file too.
	 *
	 * @return array
	 */
	protected function get_block_type_supports() {
		return array(
			'color'                  =>
			array(
				'gradients'                       => true,
				'background'                      => true,
				'link'                            => false,
				'text'                            => true,
				'__experimentalSkipSerialization' => true,
			),
			'typography'             =>
			array(
				'fontSize'                    => true,
				'lineHeight'                  => true,
				'__experimentalFontWeight'    => true,
				'__experimentalTextTransform' => true,
				'__experimentalFontFamily'    => true,
			),
			'spacing'                =>
			array(
				'margin'                          => true,
				'__experimentalSkipSerialization' => true,
			),
			'__experimentalSelector' => '.wc-block-components-product-title',
		);
	}

}
