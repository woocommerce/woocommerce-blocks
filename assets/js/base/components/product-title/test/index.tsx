/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import ProductTitle from '..';

describe( 'ProductTitle', () => {
	test( 'should render a span when heading level is false', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ false }
				showProductLink={ false }
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render an H2 when heading level is 2', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ 2 }
				showProductLink={ false }
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should not render a link if showProductLink is false', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ false }
				showProductLink={ false }
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a link if showProductLink is true', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ false }
				showProductLink={ true }
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render custom class name', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ false }
				showProductLink={ true }
				className="lorem-ipsum"
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render correct link target', () => {
		const component = TestRenderer.create(
			<ProductTitle
				headingLevel={ false }
				showProductLink={ true }
				product={ {
					id: 1,
					name: 'Test Product',
					permalink: '/',
				} }
				linkTarget="_blank"
			/>
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
