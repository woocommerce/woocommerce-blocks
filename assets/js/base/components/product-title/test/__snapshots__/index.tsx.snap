// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should not render a link if showProductLink is false 1`] = `
<span
  className="wc-block-components-product-title"
  style={Object {}}
>
  <span
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
  />
</span>
`;

exports[`ProductTitle should render a link if showProductLink is true 1`] = `
<span
  className="wc-block-components-product-title"
  style={Object {}}
>
  <a
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
    href="/"
  />
</span>
`;

exports[`ProductTitle should render a span when heading level is false 1`] = `
<span
  className="wc-block-components-product-title"
  style={Object {}}
>
  <span
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
  />
</span>
`;

exports[`ProductTitle should render an H2 when heading level is 2 1`] = `
<h2
  className="wc-block-components-product-title"
  style={Object {}}
>
  <span
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
  />
</h2>
`;

exports[`ProductTitle should render correct link target 1`] = `
<span
  className="wc-block-components-product-title"
  style={Object {}}
>
  <a
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
    href="/"
    target="_blank"
  />
</span>
`;

exports[`ProductTitle should render custom class name 1`] = `
<span
  className="lorem-ipsum wc-block-components-product-title"
  style={Object {}}
>
  <a
    className="wc-block-components-product-name"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test Product",
      }
    }
    href="/"
  />
</span>
`;
